service/storage_service.cc

```diff
@@ -1455,6 +1448,13 @@ future<> storage_service::drain_on_shutdown() {
             }).get();
             slogger.info("Drain on shutdown: hints manager is stopped");
 
+            tracing::tracing::tracing_instance().invoke_on_all([] (auto& tr) {
+                return tr.shutdown();
+            }).get();
+
+            tracing::tracing::tracing_instance().stop().get();
+            slogger.info("Drain on shutdown: tracing is stopped");
+
```

Comment:
OK for me, but two suggestions (or questions).

1. Why not stop the tracing in main.cc at the very end, in the place where it's started.
2. The invoke_in_all(tracing::stop()) is called by seastar on sharded<tracing>.stop(), why not merge the tracing.shutdown() into it?

-- Pavel

Reply:
It might work, but can be done as follow-up, possibly in the context of your shutdown cleanup project.

Reply:
I don't want to cause too much code churn as this is candidate for backporting

Reply:
Sure! I'll shuffle the code myself then.

Reply:
> OK for me, but two suggestions (or questions).
> 
>     1. Why not stop the tracing in main.cc at the very end, in the place where it's started.
> 
>     2. The invoke_in_all(tracing::stop()) is called by seastar on sharded.stop(), why not merge the tracing.shutdown() into it?
> 
> 
> -- Pavel

@xemul Please, see the description of the patch below

```
d3960f0bbbcd4a8a892ee9f233fb8e77617bf58a "tracing: rearrange shut down"
```

Reply:
@bhalevy FYI ^^^